packages/php-wasm/node/src/lib/xdebug/with-xdebug.ts

```diff
@@ -0,0 +1,70 @@
+import type {
+	EmscriptenOptions,
+	PHPRuntime,
+	SupportedPHPVersion,
+} from '@php-wasm/universal';
+import { LatestSupportedPHPVersion } from '@php-wasm/universal';
+import { jspi } from 'wasm-feature-detect';
+import { fullyQualifiedPHPVersionDirectory } from './supported-php-versions';
+import { dirname } from 'path';
+import fs from 'fs';
+
+export async function withXdebug(
+	version: SupportedPHPVersion = LatestSupportedPHPVersion,
+	options: EmscriptenOptions
+): Promise<EmscriptenOptions> {
+	if (!(await jspi())) {
+		throw new Error('Xdebug is currently only supported in JSPI mode.');
+	}
+
+	const fileName = 'xdebug.so';
+	const directoryName = fullyQualifiedPHPVersionDirectory(version);
+	const filePath = `${dirname(
+		__filename
+	)}/jspi/${directoryName}/extensions/${fileName}`;
+	const extension = fs.readFileSync(filePath);
+
+	return {
+		...options,
+		ENV: {
+			...options.ENV,
+			PHP_INI_SCAN_DIR: '/internal/shared/extensions',
```

Comment:
lovely, thank you! CC @brandonpayton as that's another directory all the workers need to share